conformance-tests/storage/retry/build-and-push-image.bash

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+set -o errexit -o nounset -o pipefail
+
+__SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+
+function main() { (
+  docker_image=$1
+
+  cd "$__SCRIPT_DIR"
+
+  short_sha="${SHORT_SHA:-${BASE_TAG}}"
+
+  clone_args=""
+  if [[ "${CPP_COMMITISH}" != "default" ]]; then
+    clone_args="--branch $CPP_COMMITISH"
+  fi
+
+  msg "cloning google-cloud-cpp repo..."
+  git clone $clone_args https://github.com/googleapis/google-cloud-cpp.git google-cloud-cpp
```

Comment:
Looking it again.

I think it's better to do these stuff in Dockerfile. WDYT?

Reply:
Then you can simplify the process, likely you can remove this file.

Reply:
Performing the clone inside the Dockerfile would create another layer in the resulting image with the entire repo, whereas we only need the specific subtree. Cloning outside the built image ensures the only the subtree is added.

We would still need this script to do the SHA calculation and the tag generation since cloud build doesn't allow setting a substitutable from within the build, only at build start time.

Reply:
ok